src/pytorch_lightning/core/module.py

```diff
@@ -1563,10 +1563,7 @@ def lr_scheduler_step(self, scheduler, optimizer_idx, metric):
                 scheduler.step(epoch=self.current_epoch)
 
         """
-        if metric is None:
-            scheduler.step()
-        else:
-            scheduler.step(metric)
+        scheduler.step(metric)
```

Comment:
@rohitgr7 will this change break anything? mypy is complaining about ln: 1567 of the previous implementation

Reply:
not always.. currently it should be passed only when user is using `ReduceLROnPlateau`.

For others the signature can vary: https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html

Reply:
reverted to original and ignored